File: backend/app/schemas/pending_memory.py
from datetime import datetime

from pydantic import BaseModel


class PendingMemoryOut(BaseModel):
    id: int
    novel_id: int
    chapter_id: int
    memory_type: str
    content: dict
    status: str
    created_at: datetime


class PendingMemoryBatchBody(BaseModel):
    ids: list[int]
    action: str  # "confirm" | "reject"


class ExtractResponse(BaseModel):
    chapter_summary: str | None = None
    pending_ids: list[int] = []
    pending_count: int = 0


class ExtractRequest(BaseModel):
    mode: str = "standard"  # "standard" | "deep"

File: frontend/src/types/pendingMemory.ts
export interface PendingMemory {
  id: number
  novel_id: number
  chapter_id: number
  memory_type: 'character_change' | 'plot_fact' | 'world_setting' | 'foreshadowing'
  content: Record<string, unknown>
  status: 'pending' | 'confirmed' | 'rejected'
  created_at: string
}

export interface ExtractRequest {
  mode: 'standard' | 'deep'
}

export interface ExtractResponse {
  chapter_summary: string | null
  pending_ids: number[]
  pending_count: number
}
